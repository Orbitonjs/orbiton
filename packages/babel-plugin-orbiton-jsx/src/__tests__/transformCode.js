/**
 * Copyright (c) 2021 - present Beignana Jim Junior and other contributors.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

/* eslint-disable no-undef */
import TransformSyntax from "../index";

const babel = require('@babel/core');
const code = `
function App(props) {
  return (
    <>
      <div:Head className="orbiton-root" props={{
        title: "Home"
      }}>
        <Route path="/button">
          <button
            onClick={(e) => {
              console.log("Clicked")
            }}
            onKeyUpCapture={activatePlayer}
          >
            Click Me
          </button>
        </Route>
        <Route>
          <svg
            xmlns="http://www.w3.org/2000/svg"
            xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
          >
            <path d="M0 0h24v24H0V0z" fill="none" />
          </svg>
          <Icons.MdSearch />
        </Route>
      </div:Head>
    </>
  )
}


`

it('Compiles JSX right', () => {
  const output = babel.transformSync(code, {
    filename: '.',
    presets: [
      [
        "@babel/preset-env",
        {
          targets: {
            node: 'current'
          }
        }
      ],
      "@babel/preset-typescript"
    ],
    plugins: [
      "@babel/plugin-syntax-jsx",
      TransformSyntax
    ],
  })
  expect(output.code).toMatchSnapshot()
});

