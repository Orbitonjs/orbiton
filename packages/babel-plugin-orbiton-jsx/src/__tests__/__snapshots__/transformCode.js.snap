// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compiles JSX right 1`] = `
"\\"use strict\\";

function App(props) {
  return Orbiton.createComponent(Orbiton.Fragment, {
    children: [Orbiton.withComponent(\\"div\\", Head, {
      attributes: {
        className: \\"orbiton-root\\"
      },
      props: {
        title: \\"Home\\"
      },
      events: {},
      children: [Orbiton.createComponent(Route, {
        path: \\"/button\\",
        children: [Orbiton.createElement(\\"button\\", {
          attributes: {},
          events: {
            click: e => {
              console.log(\\"Clicked\\");
            },
            keyupcapture: activatePlayer
          },
          children: [\\"\\\\n            Click Me\\\\n          \\"]
        })]
      }), Orbiton.createComponent(Route, {
        children: [Orbiton.createElement(\\"svg\\", {
          attributes: {
            xmlns: \\"http://www.w3.org/2000/svg\\",
            \\"xmlns:inkscape\\": \\"http://www.inkscape.org/namespaces/inkscape\\"
          },
          events: {},
          children: [Orbiton.createElement(\\"path\\", {
            attributes: {
              d: \\"M0 0h24v24H0V0z\\",
              fill: \\"none\\"
            },
            events: {},
            children: []
          })]
        }), Orbiton.createComponent(Icons.MdSearch, {
          children: []
        })]
      })]
    })]
  });
}"
`;
